src/models/event.ts

```diff
@@ -59,10 +59,7 @@ export default class Event extends RestfulModel {
 
   constructor(connection: NylasConnection, props?: EventProperties) {
     super(connection, props);
-    if (!props) {
-      this.calendarId = '';
-      this.when = new When();
-    }
+    this.initAttributes(props);
```

Comment:
The reason why all the classes have this `initAttributes` call instead of having it in the super class constructor (as it was earlier) is due to how TypeScript is designed in regards to how a child class is initialized. More info can be found here (thanks to @ozsivanov for finding this!): https://stackoverflow.com/questions/43595943/why-are-derived-class-property-values-not-seen-in-the-base-class-constructor